src/variants/service/VariantsService.ts

```diff
@@ -58,10 +59,18 @@ export default class VariantsService {
     return deferred.promise as any;
   }
 
+  public getPage(variantsSearchRequestByProductFilterId: VariantsSearchRequestByProductFilterId):
+    Promise<VariantsPage> {
+    const deferred: Q.Deferred<VariantsPage> = Q.defer<VariantsPage>();
+    const getSinglePage = true;
+    this.getVariantSearchResultsByProductFilterId(variantsSearchRequestByProductFilterId, [], deferred, getSinglePage);
+    return deferred.promise as any;
+  }
+
   private getVariantSearchResultsByRequest(
     variantSearchRequest: GenericVariantsSearchRequest,
     variantSearchDetails: VariantSearchDetails[],
-    deferred: Q.Deferred<VariantSearchDetails[]>,
+    deferred: Q.Deferred<any>,
```

Comment:
Try to avoid using any when possible. Can you just split the types of deferred there to the the two types that you need?